src/backend/pipeline/tuplebuf.c

```diff
@@ -614,7 +614,7 @@ TupleBufferTryWait(TupleBufferReader *reader)
 	SpinLockRelease(&buf->mutex);
 
 	if (should_wait)
-		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 0);
+		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 1);
```

Comment:
I just ran that loop with 32 procs and the cpu usage is negligible. How about we increase to 10ms and only wait on combiners?